bin/cli.js

```diff
@@ -28,9 +28,14 @@ const fsPromised = {
   writeFile: promisify(fs.writeFile),
 };
 
-function openKnexfile(configPath) {
-  const config = require(configPath);
-
+async function openKnexfile(configPath) {
+  let config = require(configPath);
+  if (typeof config  === 'function') {
+    config = config();
+  }
+  if (config instanceof Promise) {
```

Comment:
This is redundant, you can simply await config() and if it's not a promise, it will just return the value.